src/plugins/embeddable/public/lib/embeddables/compatibility/edit_legacy_embeddable.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Subscription } from 'rxjs';
+import { LegacyCompatibleEmbeddable } from '../../../embeddable_panel/types';
+import { core, embeddableStart } from '../../../kibana_services';
+import { Container } from '../../containers';
+import { EmbeddableFactoryNotFoundError } from '../../errors';
+import { EmbeddableEditorState } from '../../state_transfer';
+import { isExplicitInputWithAttributes } from '../embeddable_factory';
+import { EmbeddableInput } from '../i_embeddable';
+
+const getLatestAppId = async (): Promise<string | undefined> => {
```

Comment:
Could this be simplified to just `return core.application.currentAppId$.pipe(first()).toPromise();`

Reply:
Actually, toPromise is deprecated. instead, you could just use `await lastValueFrom(core.application.currentAppId$)`

Reply:
I'm not sure that `lastValueFrom` will work properly in this case. I think it waits for the observable to complete.

Reply:
Good call, `firstValueFrom` works perfectly here and makes the code much shorter!